src/main/kotlin/app/revanced/patches/shared/settings/preference/impl/TextPreference.kt

```diff
@@ -13,17 +13,17 @@ import org.w3c.dom.Element
  * @param key The key of the text preference.
  * @param title The title of the text preference.
  * @param inputType The input type of the text preference.
- * @param default The default value of the text preference.
  * @param summary The summary of the text preference.
+ * @param default The default value of the text preference.
  */
 internal class TextPreference(
     key: String,
     title: StringResource,
-    var inputType: InputType = InputType.STRING,
+    summary: StringResource,
```

Comment:
Seems like some classes that extend of `BasePreference` do not propagate the correct type. Some classes have the `summary` of type `StringResource` and some of type `StringResource?` (nullable), whereas you might not always accompany a `TextPreference` with a summary for example.

Reply:
> Some existing code doesn't have a summary for the items.

Can you clarify what you mean by that?

Reply:
> SwitchPreference has no single summary and uses two (summary switch on, summary switch off)

That seems to me that we have a little issue with our abstraction. 
`SwitchPreference` extends to `BasePreference`, which implements `IPreference` with the `summary` property, but `SwitchPreference` does not have `summary` thus violating Liskov substitution.

Reply:
I'll push some changes in a minute.